contracts/registries/IPAssetRegistry.sol

```diff
@@ -80,6 +86,19 @@ contract IPAssetRegistry is
     }
 
     function _register(uint256 chainid, address tokenContract, uint256 tokenId) internal override returns (address id) {
+        IPAssetRegistryStorage storage $ = _getIPAssetRegistryStorage();
+
+        // Pay registration fee
+        uint96 feeAmount = $.feeAmount;
+        if (feeAmount > 0) {
+            address feeToken = $.feeToken;
+            address treasury = $.treasury;
+            if (feeToken != address(0) && treasury != address(0)) {
```

Comment:
`if (feeToken != address(0) && treasury != address(0))` -> Do we need this? Isn't this mandatory given the code in `setRegistrationFee` ? Or could I be missing something?

if (feeAmount > 0) {
            if (treasury == address(0)) revert Errors.IPAssetRegistry__ZeroAddress("treasury");
            if (feeToken == address(0)) revert Errors.IPAssetRegistry__ZeroAddress("feeToken");
        }

Reply:
Yeah, it is safe to remove this check.  

resolved with commit ee1f4a4